staging/src/k8s.io/apiserver/pkg/endpoints/request/requestcontext.go

```diff
@@ -55,20 +55,21 @@ func NewRequestContextMapper() RequestContextMapper {
 // Get returns the context associated with the given request (if any), and true if the request has an associated context, and false if it does not.
 // Get will only return a valid context when called from inside the filter chain set up by NewRequestContextFilter()
 func (c *requestContextMap) Get(req *http.Request) (Context, bool) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
+	c.lock.RLock()
+	defer c.lock.RUnlock()
 	context, ok := c.contexts[req]
 	return context, ok
 }
 
 // Update maps the request to the given context.
 // If no context was previously associated with the request, an error is returned and the context is ignored.
 func (c *requestContextMap) Update(req *http.Request, context Context) error {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	if _, ok := c.contexts[req]; !ok {
+	if _, ok := c.Get(req); !ok {
 		return errors.New("No context associated")
 	}
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
```

Comment:
reverify it does not exist after locking

Reply:
make sense, like what we do in crd